chainer/functions/connection/n_step_rnn.py

```diff
@@ -303,8 +327,19 @@ def forward(self, inputs):
         # TODO(unno): Make a wrapper method to avoid access _desc directly
         rnn_desc = cudnn.create_rnn_descriptor(
             n_units, self.n_layers, self.states._desc,
-            libcudnn.CUDNN_LINEAR_INPUT, self.rnn_dir,
-            self.rnn_mode, libcudnn.CUDNN_DATA_FLOAT)
+            libcudnn.CUDNN_LINEAR_INPUT, self.rnn_dir, self.rnn_mode,
+            libcudnn.CUDNN_DATA_FLOAT, algo=self.rnn_algo)
+
+        if self.rnn_algo == libcudnn.CUDNN_RNN_ALGO_PERSIST_DYNAMIC:
+            batchsize = len(x_list[0])
+            global _prev_batchsize
```

Comment:
Is it thread safe?

Reply:
Do you have any idea to cache `_rnn_plan ` when a batchsize is same?